Makefile

```diff
@@ -14,32 +18,35 @@ BUILDPACK_API?=0.2
 SCM_REPO?=github.com/buildpacks/lifecycle
 PARSED_COMMIT:=$(shell git rev-parse --short HEAD)
 SCM_COMMIT?=$(PARSED_COMMIT)
+PWD?=${CURDIR}
 BUILD_DIR?=$(PWD)/out
-COMPILATION_IMAGE?=golang:1.13-alpine
-
-define LIFECYCLE_DESCRIPTOR
-[api]
-  platform = "$(PLATFORM_API)"
-  buildpack = "$(BUILDPACK_API)"
-
-[lifecycle]
-  version = "$(LIFECYCLE_VERSION)"
-endef
+LINUX_COMPILATION_IMAGE?=golang:1.13-alpine
+WINDOWS_COMPILATION_IMAGE?=golang:1.14-windowsservercore-1809
+SOURCE_COMPILATION_IMAGE?=lifecyle-img
+BUILD_CTR?=lifecycle-ctr
+DOCKER_CMD?=make test
 
 all: test build package
 
 build: build-linux build-windows
 
+build-linux: build-linux-lifecycle build-linux-symlinks build-linux-launcher
+
+ifeq ($(OS),Windows_NT)
```

Comment:
Nearly all the make targets can run on Windows/Linux/MacOS. `build-windows` is the only one required for CI that has to switch currently